vendor/engines/secure_rooms/app/services/secure_rooms/access_handlers/order_handler.rb

```diff
@@ -0,0 +1,54 @@
+module SecureRooms
+
+  module AccessHandlers
+
+    class OrderHandler
+
+      attr_reader :occupancy, :order, :order_detail
+
+      def self.process(occupancy)
+        new(occupancy).process
+      end
+
+      def initialize(occupancy)
+        @occupancy = occupancy
+      end
+
+      def process
+        return unless occupancy.account_id?
+
+        create_order
```

Comment:
We're doing a bunch of DB work here. I think it might be a good idea to wrap all this inside a transaction.